testing/time.ts

```diff
@@ -373,7 +349,7 @@ export class FakeTime {
    * ```
    */
   static restore() {
-    if (!time) throw new TimeError("Time is already restored");
+    if (!time) throw new TypeError("Time is already restored");
```

Comment:
Note: made to be similar to when a constant tried to get re-assigned ([ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_const_assignment)).

Reply:
`TypeError` looks strange to me here because this is more related to 'state' of fake time, not 'type'.

Reply:
Updated to `Error`.

Reply:
Well, another way to think about it is you're trying to change `time` to be `undefined`, when it can only be changed to be defined. Does that make sense?

Reply:
`time` variable is implementation detail and not visible to the users.